hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/RegionReplicaReplicationEndpoint.java

```diff
@@ -289,7 +289,7 @@ public boolean replicate(ReplicateContext replicateContext) {
           }
           continue outer;
         }
-        if (!requiresReplication(tableDesc, entry)) {
+        if (!requiresReplication(tableDesc.orElse(null), entry)) {
```

Comment:
OK, again, here we unwrap the optional by checking whether it is null and inside the method we checl it again...

Where does this Optional come from? If we could not change the root, let's keep it as is?